cli/src/commands/config/unset.rs

```diff
@@ -12,21 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use clap_complete::ArgValueCandidates;
 use tracing::instrument;
 
 use super::ConfigLevelArgs;
 use crate::cli_util::get_new_config_file_path;
 use crate::cli_util::CommandHelper;
 use crate::command_error::user_error;
 use crate::command_error::CommandError;
+use crate::complete;
 use crate::config::remove_config_value_from_file;
 use crate::config::ConfigNamePathBuf;
 use crate::ui::Ui;
 
 /// Update config file to unset the given option.
 #[derive(clap::Args, Clone, Debug)]
 pub struct ConfigUnsetArgs {
-    #[arg(required = true)]
+    #[arg(required = true, add = ArgValueCandidates::new(complete::config_keys))]
```

Comment:
maybe this should also be a "leaf_config_keys"?

Reply:
I checked, and unset seems to work with objects as well. If I do:
```sh
jj config unset --user aliases
```
The entire `aliases` table is removed from my config. While it could be a bit of a footgun, the current behavior makes sense I think.

Reply:
Indeed. I believe it's a bug. The comment says "remove config value".

Reply:
Alright, I'll change it to `leaf_config_keys` in that case. I'll leave fixing the bug to someone else :slightly_smiling_face: